torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3827,28 +3846,47 @@ struct to_ir {
   Value* emitTupleSlice(
       const SourceRange& loc,
       const NamedValue& tuple_val,
-      const NamedValue& beg_val,
-      const at::optional<NamedValue>& end_val) {
+      const at::optional<NamedValue>& beg_val,
+      const at::optional<NamedValue>& end_val,
+      const at::optional<NamedValue>& step) {
     auto tuple_type = tuple_val.value(*graph)->type()->expect<TupleType>();
-    int64_t beg = getAdjTupleIndex(
-        loc,
-        tuple_type,
-        getSliceInd(beg_val.value(*graph), loc),
-        /*allow_out_of_bounds*/ true);
-    int64_t end;
     int64_t tuple_len = tuple_type->elements().size();
+    // first figure out the step sign to know if we want to flip the start or
+    // end. For example, if we have x[::-3], the start should be len(x) - 1 and
+    // end should be 0
+    int64_t step_size = 1;
+    if (step) {
+      auto val = toIValue(step->value(*graph));
+      if (val->isInt()) {
```

Comment:
I think it is safe to `TORCH_CHECK(val->isInt())` here. Otherwise, step is silently ignored if a non-int value is provided.

Reply:
sounds good!